linux/LinuxProcess.c

```diff
@@ -195,6 +204,303 @@ void LinuxProcess_printDelay(float delay_percent, char* buffer, int n) {
 }
 #endif
 
+/*
+Note that when colorizing a basename with the comm prefix, the entire basename
+(not just the comm prefix) is colorized for better readability, and it is
+implicit that only upto (TASK_COMM_LEN - 1) could be comm
+*/
+static inline bool findCommInCmdline(const char *comm, char *cmdline, int cmdlineBasenameOffset, int *pCommStart, int *pCommEnd) {
```

Comment:
This suggestion is a bit incorrect, as it would successfully match token == "abc" against comm == "abcd" - but it shouldn't. Still, if we wish to anyway avoid modifying (inserting/restoring NUL terminators in) cmdline, one way will be to find the commLen = strlen(comm), and then use a logic such as this:
`
      tokenLen = token - tokenBase;

      if (((commLen < (TASK_COMM_LEN - 1) && tokenLen == commLen) ||
	   (commLen == (TASK_COMM_LEN - 1) && tokenLen >= commLen)) &&
	  strncmp(tokenBase, comm, commLen) == 0) {
         *pCommStart = tokenBase - cmdline;
         *pCommEnd = token - cmdline;
         return true;
      }
`